### PR TITLE
feat: add some config flags with bitflags in chapter 11

### PR DESCRIPTION
**Pull request by GrassedgeT**
Add  some config flags with bitflags crate in chapter 11

These flags declared in: 

Table 36(PMU Counter Config Match Flags),
Table 38(PMU Counter Start Flags),
Table 40(PMU Counter Stop Flags).

### CODE REVIEW

**Comment by luojia65** [comment]
Thanks for your contribution! I'll look into it, but I may have better options for the module name `counter_related`. What about the name `flags`?

File: sbi-spec/src/pmu.rs [hunk 1]
```diff
@@ -185,3 +185,57 @@ pub mod firmware_event {
     /// The `event_data` configuration (or parameter) contains the event encoding.
     pub const PLATFORM: usize = 65535;
 }
+
+/// Find and configure a matching counter.
+/// Start a set of counters.
+/// Stop a set of counters.
+///  
+/// Declared in §11.7 §11.8 §11.9.
```
Comment (luojia65) [inline sbi-spec/src/pmu.rs:193]:
```suggestion
/// Declared in §11.7, §11.8 and §11.9.
```

File: sbi-spec/src/pmu.rs [hunk 2]
```diff
@@ -185,3 +185,57 @@ pub mod firmware_event {
...
+pub mod counter_related {
+    use bitflags::bitflags;
+
+    bitflags! {
+        #[derive(Clone, Copy, PartialEq, Eq)]
+        /// Declared in Table 36.
+        pub struct CounterCfgFlags: usize{
```
Comment (luojia65) [inline sbi-spec/src/pmu.rs:200]:
```suggestion
        pub struct CounterCfgFlags: usize {
```

File: sbi-spec/src/pmu.rs [hunk 3]
```diff
@@ -185,3 +185,57 @@ pub mod firmware_event {
...
+            const INIT_VALUE = 1 << 0;
+            /// Initialize the given counters from shared memory if available.
+            const INIT_SNAPSHOT = 1 << 1;
+        }
+    }
+
+    bitflags! {
+        #[derive(Clone, Copy, PartialEq, Eq)]
+        /// Declared in Table 40.
+        pub struct CounterStopFlags: usize{
```
Comment (luojia65) [inline sbi-spec/src/pmu.rs:234]:
```suggestion
        pub struct CounterStopFlags: usize {
```

File: sbi-spec/src/pmu.rs [hunk 4]
```diff
@@ -185,3 +185,57 @@ pub mod firmware_event {
...
+            const SET_SINH = 1 << 6;
+            /// Event counting inhibited in M-mode.
+            const SET_MINH = 1 << 7;
+        }
+    }
+
+    bitflags! {
+        #[derive(Clone, Copy, PartialEq, Eq)]
+        /// Declared in Table 38
+        pub struct CounterStartFlags: usize{
```
Comment (luojia65) [inline sbi-spec/src/pmu.rs:223]:
```suggestion
        pub struct CounterStartFlags: usize {
```

**Review by luojia65 — CHANGES_REQUESTED** [review]

**Comment by GrassedgeT** [comment]
Thanks for your contribution! I'll look into it, but I may have better options for the module name `counter_related`. What about the name `flags`?
Translated Text: >
BTW, please add a changelog entry for your contribution.

Ok, I have limited experience in contributing, thanks for the advice.

**Comment by luojia65** [comment]
I'll review recently, thanks!

**Review by luojia65 — APPROVED** [review]
LGTM, I'll adjust documents and modify those structure names later.

**Pull request merged**